client/lib/signup/step-actions/index.js

```diff
@@ -811,6 +811,49 @@ export function createAccount(
 	}
 }
 
+function postToCreateSite( data, reduxStore, callback ) {
+	wpcom.req.post( '/sites/new', data, function ( errors, response ) {
+		let providedDependencies;
+		let siteSlug;
+
+		if ( response && response.blog_details ) {
+			const parsedBlogURL = getUrlParts( response.blog_details.url );
+			siteSlug = parsedBlogURL.hostname;
+
+			providedDependencies = { siteSlug };
+		}
+
+		if ( isUserLoggedIn( reduxStore.getState() ) && isEmpty( errors ) ) {
+			fetchSitesAndUser( siteSlug, () => callback( undefined, providedDependencies ), reduxStore );
+		} else {
+			callback( isEmpty( errors ) ? undefined : [ errors ], providedDependencies );
+		}
+	} );
+}
```

Comment:
We need to be sure this refactor doesn't cause any regressions. It shouldn't, but I wanted to call it out for review.